pkg/cluster/deploystorage_resources.go

```diff
@@ -74,16 +74,75 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
-func (m *manager) storageAccount(name, region string) *arm.Resource {
-	return &arm.Resource{
-		Resource: &mgmtstorage.Account{
-			Sku: &mgmtstorage.Sku{
-				Name: "Standard_LRS",
+// storageAccount will return storage account resource.
+// Old accounts are not encrypted and can't be retrofitted.
+// flag is to controll this behaviour in update/create
+func (m *manager) storageAccount(name, region string, encrypted bool) *arm.Resource {
+	sa := &mgmtstorage.Account{
+		Kind: mgmtstorage.StorageV2,
+		Sku: &mgmtstorage.Sku{
+			Name: "Standard_LRS",
+		},
+		AccountProperties: &mgmtstorage.AccountProperties{
+			AllowBlobPublicAccess:  to.BoolPtr(false),
+			EnableHTTPSTrafficOnly: to.BoolPtr(true),
+			MinimumTLSVersion:      mgmtstorage.TLS11,
```

Comment:
Note that from 4.7 the installer is using TLS 1.2 as `MinimumTLSVersion` for the cluster storrage accounts in `public` and `usgovernment` clouds and default value (which is TLS 1.0 I believe) in other clouds.

I think we need to do the same.

https://github.com/openshift/installer/blob/422e17192462f2b5d5572d8600096949d3c14634/data/data/azure/vnet/main.tf#L55

Reply:
I created https://github.com/Azure/ARO-RP/pull/1730 to go with 4.8.